earth_enterprise/tutorial/SConstruct

```diff
@@ -34,30 +34,62 @@ if (not installdir) and ('install' in COMMAND_LINE_TARGETS):
-    # get contents of target_dir and make hard links to files in instdir
+# Get the current directory, since SCons will move into the directory
+#   it's using, and FusionTutorial is expected below the SConscript file.
+tutorial_src_dir = os.sep.join((os.getcwd().split(os.sep) +
+                    ['FusionTutorial']))
+
+def makeHardLinks(install_dir, target_dir, excluded_list):
+    # get contents of target_dir and make hard links to files in install_dir
     # recurse down the directories
 
     contents = os.listdir(target_dir)
-    target_dir = target_dir + "/"
+    target_dir = target_dir + "/"	
     # Ensure the install dir exists at each step
-    commands.getstatusoutput("mkdir -p " + instdir)
+    commands.getstatusoutput("mkdir -p " + install_dir)
     for file in contents:
         if file in excluded_list:
             continue
         target_file = target_dir + file
         if os.path.isdir(target_file):
-            subdir = instdir + "/" + file
+            subdir = install_dir + "/" + file
             makeHardLinks(subdir, target_file,
                                              excluded_list)
         else:
             commands.getstatusoutput("ln \"" + target_file + "\" " +
-                                     instdir)
+                                     install_dir)
+
+
+def makeSoftLinks(dest_dir, src_dir, excluded_list):
+    """Make soft links of all files in src_dir and place
+    them in dest_dir. Don't link to files in excluded_list.
+    Recurse through all subdirectories of src_dir.
+    """
+    # Strip trailing slashes because they cause trouble later
+    dest_dir = dest_dir.rstrip(os.sep)
+    src_dir = src_dir.rstrip(os.sep)
+    src_contents = os.listdir(src_dir)
+    # Ensure the destination directory exists at each step
+    commands.getstatusoutput("mkdir -p " + dest_dir)
+    for file in src_contents:
+        if file in excluded_list:
+            continue
+        # Use full paths to all files so `ln -s ... ` succeeds 
+        dest_filepath = os.sep.join((dest_dir.split(os.sep) + [file]))
+        src_filepath = os.sep.join((src_dir.split(os.sep) + [file]))
+        if os.path.isdir(src_filepath):
+            dest_subdir = os.sep.join((dest_dir.split(os.sep) + [file]))
+            src_subdir = os.sep.join((src_dir.split(os.sep) + [file]))
+            # Recurse for next-lower directory
+            makeSoftLinks(dest_subdir, src_subdir, excluded_list)
+        else:
+            commands.getstatusoutput('ln -s "' + src_filepath + '" "' +
+                                    dest_filepath + '"')
 
 # Clean install dir (note: the current sconscripts leave stuff behind, like
 # directories).
 if env.GetOption('clean'):
     # clean the search and getomcat install folders.
     commands.getstatusoutput("rm -rf " + tutorials_dir)
 else:
-    makeHardLinks(root_dir, 'FusionTutorial/', ['.svn'])
+    makeSoftLinks(root_dir, tutorial_src_dir, ['.svn'])
```

Comment:
Should we check to see if hard-links are possible and prefer them if they are?  Then only use soft-links if we have to.  Otherwise we should delete the dead code.